app/module/test-behaviors/behavior_add_layer.c

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2025 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#define DT_DRV_COMPAT zmk_behavior_add_layer
+
+#include <zephyr/device.h>
+#include <drivers/behavior.h>
+#include <zephyr/logging/log.h>
+
+#include <zmk/keymap.h>
+#include <zmk/behavior.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+#if (IS_ENABLED(CONFIG_ZMK_KEYMAP_LAYER_REORDERING)) && (DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT))
+
+static int on_add_layer_binding_pressed(struct zmk_behavior_binding *binding,
+                                        struct zmk_behavior_binding_event event) {
+    int new_layer = zmk_keymap_add_layer();
+    switch (new_layer) {
+    case -ENOSPC:
+        LOG_ERR("No more layers can be added");
+        return -ENOSPC;
+    default:
+        LOG_DBG("Added layer %d", new_layer);
+        return 0;
+    }
```

Comment:
To be future proof, this should really only consider it success if `new_layer >= 0`, in case other errno start getting returned from the called API down the road.